nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -132,37 +130,39 @@ def _get_spheres_rows(seeds, radius, allow_overlap, mask_img):
             nearests.append(None)
 
     brain_coords = np.asarray(list(zip(*brain_coords)))
-    # Calculate mask
+    # Calculate mask and the corresponding index in flattended array.
     mask_coords = list(zip(*np.where(mask == 1)))
     mask_coords = np.asarray(mask_coords).astype(int).T
     mask_idx = np.vstack([np.in1d(brain_coords[0, :], mask_coords[0, :]),
```

Comment:
Possible source of un-expected behavior 1: 
I am calculating the overlapping coordinates between the whole brain and the mask incorrectly. 
Or I am incorrectly extracting the mask coordinates to begin with.